modules/server_new/src/main/scala/observe/server/SequenceData.scala

```diff
@@ -11,14 +11,36 @@ import observe.model.SystemOverrides
 import observe.model.enums.PendingObserveCmd
 
 case class SequenceData[F[_]](
-  observer:      Option[Observer],
-  overrides:     SystemOverrides,
-  seqGen:        SequenceGen[F],
-  seq:           Sequence.State[F],
-  pendingObsCmd: Option[PendingObserveCmd]
-)
+  observer:       Option[Observer],
+  overrides:      SystemOverrides,
+  seqGen:         SequenceGen[F],
+  seq:            Sequence.State[F],
+  pendingObsCmd:  Option[PendingObserveCmd],
+  visitStartDone: Boolean,
+  atomStartDone:  Boolean
+) {
+  def completeVisitStart: SequenceData[F] = this.copy(visitStartDone = true)
```

Comment:
Since these methods are not actually doing any action (completing things), may I suggest renaming to `withVisitStartDone` and `withAtomStartDone` ?